libraries/chain/include/graphene/chain/proposal_object.hpp

```diff
@@ -52,8 +52,10 @@ class proposal_object : public abstract_object<proposal_object>
       flat_set<account_id_type>     available_owner_approvals;
       flat_set<public_key_type>     available_key_approvals;
       account_id_type               proposer;
+      std::string                   fail_reason;
 
-      bool is_authorized_to_execute(database& db)const;
+      bool is_authorized_to_execute(database& db) const;
+      bool is_authorized_to_execute(database& db, std::string& fail_reason) const;
```

Comment:
I was unaware that some think that a const member function is unable to modify anything. I'm sorry that it would be confusing to some. This change will probably be backed out due to your "misunderstood intention" comment above, but I'll just make this comment to educate others:

A ``const`` member function means that member variables of the object will not be modified by the call. Variables that are not part of the object can be modified, and you can be fairly sure they will be if you're passing in a non-const reference or a non-const pointer.